src/gpustat/usr/local/emhttp/plugins/gpustat/GPUStatSettings.page

```diff
@@ -90,18 +90,25 @@ Title="GPU Statistics"
       include_once './plugins/gpustat/gpustatus.php';
     }
 
-    
-  
+
+
     $multi_enable=version_compare(parse_ini_file('/etc/unraid-version')['version'],'6.12.0-beta5', '>') ;
 
     if (count($_POST)) {
       $cfg = NULL ;
       if ($_POST['#apply'] == "_(Apply)_") {
         array_map(function($k, $v) use (&$cfg) { if($k[0] != "#") $cfg[$k] = $v; }, array_keys($_POST), $_POST );
         save_cfg($cfg,$gpustat_data) ;
-        $gpustat_cfg        = parse_plugin_cfg("gpustat", true);
-        unset($_POST) ;
-      } 
+
+        // Restart nchan gpustat if not running when refresh setting set to Yes
+        $pid = exec('pgrep gpustat');
+        if (!$pid && $_POST['UIREFRESH'] == '1' && $gpustat_cfg['UIREFRESH'] != '1') {
```

Comment:
This should be started automatically when you go back to the dashboard so not sure if this is needed. There is a file in var/run called nchan.pid which contains the running nchan processes but would need to confirm the correct way to stop a process.

Reply:
I thought about it too, but [monitor_nchan](https://github.com/unraid/webgui/blob/master/sbin/monitor_nchan) doesn't restart it if path to [gpustat ](https://github.com/SimonFair/gpustat-unraid/blob/c53a46969da24c1ca9f63357f0e0cfd2b7e0d405/src/gpustat/usr/local/emhttp/plugins/gpustat/nchan/gpustat) is present in `/var/run/nchan.pid`.
I didn't find a proper way to stop a nchan process. The only way to stop a process is by adding `:stop` in .page header to nchan attribut, but it's only for a single run. Process killed after 30 seconds.

I made change to remove process path in nchan.pid and stop process itself if refresh set to No. Dashboard will restart it every time if necessary.
Removed code from [GPUStatSettings.page](https://github.com/SimonFair/gpustat-unraid/pull/28/files/c53a46969da24c1ca9f63357f0e0cfd2b7e0d405#diff-b51fd7d3322d813ba8a13be2b2b222df3e78bb17dccfd61bef53098326f904bb) and updated here: https://github.com/SimonFair/gpustat-unraid/blob/563471bfc716cb3116ea738366af5d37aa46bb37/src/gpustat/usr/local/emhttp/plugins/gpustat/nchan/gpustat#L106-L110